compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/events/DisposableEventListener.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.events
+
+import org.w3c.dom.events.Event
+import org.w3c.dom.events.EventTarget
+
+internal fun interface DisposableEventListener {
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener:

> Event listeners can also be removed by passing an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) to an [addEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) and then later calling [abort()](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort) on the controller owning the signal.

It seems like this is what you're trying to re-implement

Reply:
Nope, this is intentional, AbortSignal has slightly different semantics and used basically for cancelling asynchronous operations (though on of consequences of it is indeed removing the event), removeEventListener is the standard way of removing events when they are no more needed. The other reason is that actually we have slightly more complicated scenarios that yet to be covered - like media query events which just won't work with AbortSignal